# 📝 Examen de Pandas Básico


## Introducción

Una empresa multinacional dedicada a la comercialización de frutas frescas opera en distintos mercados a través de sucursales físicas y canales digitales. Diariamente, la compañía registra miles de transacciones provenientes de diversas zonas geográficas y de su plataforma online.

Debido al crecimiento del negocio y al aumento en el volumen de datos, la empresa ha identificado la necesidad de mejorar su capacidad de análisis para entender mejor el comportamiento de sus ventas. En particular, busca responder preguntas clave como: qué productos generan mayores ingresos, qué zonas concentran más ventas y cómo se distribuye la demanda entre sus distintos clientes.

Para ello, el equipo de datos ha dispuesto un archivo llamado “Ventas.csv”, que contiene el detalle de las transacciones realizadas durante un día específico. Este archivo incluye información sobre la zona de venta, el cliente, el producto, la cantidad vendida y el precio unitario.

Tu rol como analista es procesar esta información utilizando Python y Pandas, con el objetivo de generar indicadores claros y un resumen que apoye la toma de decisiones comerciales.


**Archivo de referencia:** `Ventas.csv`

**Columnas:**  
`Numero,Zona,Cliente,Producto,Cantidad,Precio`

---

## Instrucciones
1. Usar Python y Pandas para resolver todas las preguntas.  
2. Guardar los resultados en variables y mostrar con `print()` cuando corresponda.  

---

# Preguntas

## 1️⃣ Leer el archivo
- Importa Pandas y carga el archivo `Ventas.csv` en un DataFrame llamado `df`.
- Muestra las primeras 7 filas.

In [1]:
import pandas as pd

df = pd.read_csv('Ventas.csv', sep=",", encoding='utf-8') #separado por comas
df.head(7)

,Numero,Zona,Cliente,Producto,Cantidad,Precio
0,1,Norte,UrbanSphere Co,Naranja,52,8
1,2,Online,EcoWave Industries,Naranja,196,8
2,3,Centro,GreenField Enterprises,Uva,435,7
3,4,Norte,Orion Ventures,Mango,641,4
4,5,Sur,NovaLink Group,Pera,48,10
5,6,Sur,ClearStream Solutions,Pera,313,10
6,7,Online,Infinity Networks,Cereza,991,2


## 2️⃣ Inspección

- Cuantas filas y columnas tiene el Dataframe.

In [2]:
(df.shape)

(1000, 6)

## 3️⃣ Selección de columnas

- Muestra solo las columnas Cliente y Producto.

In [3]:
df[['Cliente', 'Producto']]

,Cliente,Producto
0,UrbanSphere Co,Naranja
1,EcoWave Industries,Naranja
2,GreenField Enterprises,Uva
3,Orion Ventures,Mango
4,NovaLink Group,Pera
...,...,...
995,Pioneer Robotics,Manzana
996,SummitWorks Inc,Piña
997,Orion Ventures,Naranja
998,TechNova Solutions,Fresa


## 4️⃣ Filtrado de datos

- Muestra todas las ventas que se hicieron en la zona "Norte".

In [4]:
df[df['Zona']=='Norte'] #solo zona norte

,Numero,Zona,Cliente,Producto,Cantidad,Precio
0,1,Norte,UrbanSphere Co,Naranja,52,8
3,4,Norte,Orion Ventures,Mango,641,4
8,9,Norte,Infinity Networks,Banana,21,9
12,13,Norte,TechNova Solutions,Mango,604,4
13,14,Norte,Infinity Networks,Fresa,230,2
...,...,...,...,...,...,...
954,955,Norte,BluePeak Technologies,Uva,754,7
975,976,Norte,Orion Ventures,Manzana,454,2
979,980,Norte,EcoWave Industries,Manzana,50,2
985,986,Norte,BluePeak Technologies,Banana,488,9


## 5️⃣ Operaciones Básicas

- Crea una nueva columna Total con el cálculo de (Cantidad * Precio).
- Muestra el DataFrame actualizado ( 8 primeras filas)

In [5]:
df['Total'] = df['Cantidad'] * df['Precio'] #crea nueva columna con operacion en el df
df.head(8)

,Numero,Zona,Cliente,Producto,Cantidad,Precio,Total
0,1,Norte,UrbanSphere Co,Naranja,52,8,416
1,2,Online,EcoWave Industries,Naranja,196,8,1568
2,3,Centro,GreenField Enterprises,Uva,435,7,3045
3,4,Norte,Orion Ventures,Mango,641,4,2564
4,5,Sur,NovaLink Group,Pera,48,10,480
5,6,Sur,ClearStream Solutions,Pera,313,10,3130
6,7,Online,Infinity Networks,Cereza,991,2,1982
7,8,Centro,GreenField Enterprises,Fresa,145,2,290


## 6️⃣ Análisis

- ¿Cuál fué el total de ventas del día?
- ¿Cuál es la venta por cada zona?

In [6]:
# Total de ventas del día
total_ventas = df['Total'].sum() #Sumamos la nueva columna total para obtener el total de ventas del día
print('Total de ventas del día:', total_ventas)

# Venta por cada zona
ventas_por_zona = df.groupby('Zona')['Total'].sum() #Agrupamos por zona y sumamos el total de ventas
print('\nVentas por zona:')
print(ventas_por_zona)

Total de ventas del día: 2702306

Ventas por zona:
Zona
Centro    711993
Norte     644954
Online    720846
Sur       624513
Name: Total, dtype: int64


## 7️⃣ Guardar Resultados

- Generar un archivo "resumen.csv" que contenga:

    - Suma de Ventas por Zona (Descripcion y Total)
    - Suma de Ventas por producto (Descripcion y Total)
    - Cantidad de registros (transacciones) : xxxxx
    - Suma de Ventas: yyyyy

- El archivo debiera tener 3 columnas:
    - Descripcion 
    - Valor
    - Tipo

- Ejemplo:


| Descripción | Valor  | Tipo      |
|------------|-----------|-----------|
| Centro     | 711,993   | Zonas     |
...
| Sur        | 624,513   | Zonas     |
| Banana     | 429,714   | Productos |
| Cereza     | 107,314   | Productos |
...
| Piña       | 335,352   | Productos |
| Uva        | 352,107   | Productos |
| venta      | 2,702,306 | Resumen   |
| registros  | 1,000     | Resumen   |



Ayuda: Algunas maneras de renombrar Las columnas de un dataframe:

- df = df.rename(columns={'nombre_antiguo': 'nombre_nuevo'})
- df.rename(columns={"A": "Columna1", "B": "Columna2"}, inplace=True)
- df.columns = ['nuevo_nombre1', 'nuevo_nombre2', ...]

In [7]:
# Suma de ventas por zona
ventas_por_zona = df.groupby('Zona')['Total'].sum().reset_index()
ventas_por_zona = ventas_por_zona.rename(columns={'Zona': 'Descripcion', 'Total': 'Valor'})
ventas_por_zona['Tipo'] = 'Zonas'
ventas_por_zona


,Descripcion,Valor,Tipo
0,Centro,711993,Zonas
1,Norte,644954,Zonas
2,Online,720846,Zonas
3,Sur,624513,Zonas


In [8]:
# Suma de ventas por producto
ventas_por_producto = df.groupby('Producto')['Total'].sum().reset_index()
ventas_por_producto = ventas_por_producto.rename(columns={'Producto': 'Descripcion', 'Total': 'Valor'})
ventas_por_producto['Tipo'] = 'Productos'
ventas_por_producto

,Descripcion,Valor,Tipo
0,Banana,429714,Productos
1,Cereza,107314,Productos
2,Fresa,105436,Productos
3,Kiwi,166965,Productos
4,Mango,169372,Productos
5,Manzana,88136,Productos
6,Naranja,420880,Productos
7,Pera,527030,Productos
8,Piña,335352,Productos
9,Uva,352107,Productos


In [9]:
# Resumen general
total_ventas = df['Total'].sum()
registro_count = len(df)
resumen_general = pd.DataFrame([
    {'Descripcion': 'Venta', 'Valor': total_ventas, 'Tipo': 'Resumen'},
    {'Descripcion': 'Registros', 'Valor': registro_count, 'Tipo': 'Resumen'}
])
resumen_general

,Descripcion,Valor,Tipo
0,Venta,2702306,Resumen
1,Registros,1000,Resumen


In [10]:
# Combinar en un solo DataFrame
df_resumen = pd.concat([ventas_por_zona, ventas_por_producto, resumen_general], ignore_index=True)

#Guardar el resumen en un archivo CSV y mostrar el nuevo df_resumen
df_resumen.to_csv('Valeria-Luna-resumen.csv', index=False, sep=';',encoding='utf-8')
df_resumen

,Descripcion,Valor,Tipo
0,Centro,711993,Zonas
1,Norte,644954,Zonas
2,Online,720846,Zonas
3,Sur,624513,Zonas
4,Banana,429714,Productos
5,Cereza,107314,Productos
6,Fresa,105436,Productos
7,Kiwi,166965,Productos
8,Mango,169372,Productos
9,Manzana,88136,Productos
